CRM/Utils/Mail/EmailProcessor.php

```diff
@@ -117,34 +118,38 @@ private static function _process($civiMail, $dao, $is_create_activities) {
 
           // if its the activities that needs to be processed ..
           try {
-            $mailParams = CRM_Utils_Mail_Incoming::parseMailingObject($mail, $createContact, FALSE, $emailFields);
+            $mailParams = CRM_Utils_Mail_Incoming::parseMailingObject($mail, $incomingMail->getAttachments(), $createContact, $emailFields, [$incomingMail->getFrom()]);
             $activityParams = [
               'activity_type_id' => (int) $dao->activity_type_id,
               'campaign_id' => $dao->campaign_id ? (int) $dao->campaign_id : NULL,
               'status_id' => $dao->activity_status,
+              'subject' => $incomingMail->getSubject(),
+              'activity_date_time' => $incomingMail->getDate(),
+              'details' => $incomingMail->getBody(),
             ];
+            if ($incomingMail->isVerp()) {
+              $activityParams['source_contact_id'] = $incomingMail->lookup('Queue', 'contact_id');
+            }
+            else {
+              $activityParams['source_contact_id'] = $mailParams[$dao->activity_source][0]['id'];
 
-            $activityParams['source_contact_id'] = $mailParams[$dao->activity_source][0]['id'];
-
-            $activityContacts = ['target_contact_id' => 'activity_targets', 'assignee_contact_id' => 'activity_assignees'];
-            foreach ($activityContacts as $activityContact => $daoName) {
-              $activityParams[$activityContact] = [];
-              $activityKeys = array_filter(explode(",", $dao->$daoName));
```

Comment:
Lost this line

Reply:
@larssandergreen so `  $activityKeys = array_filter(explode(",", $dao->$daoName));` was from looking up the dao again but now it comes from the earlier lookup 

```
   $activityContacts = [
                'target_contact_id' => $targetFields,
                'assignee_contact_id' => $assigneeFields,
              ];
```

Reply:
Right, makes sense